src/shadow-embed.js

```diff
@@ -314,6 +316,40 @@ function getStylesheetRules(doc, css) {
   }
 }
 
+/**
+ * @param {!ShadowRoot} shadowRoot
+ * @param {string} name
+ * @param {string} cssText
+ */
+export function installShadowStyle(shadowRoot, name, cssText) {
+  const doc = shadowRoot.ownerDocument;
+  const win = toWin(doc.defaultView);
+  if (shadowRoot.adoptedStyleSheets !== undefined &&
+      CSSStyleSheet.prototype.replaceSync !== undefined) {
+    const cache = win[SHADOW_CSS_CACHE] || (win[SHADOW_CSS_CACHE] = {});
+    let styleSheet = cache[name];
+    if (!styleSheet) {
+      styleSheet = new CSSStyleSheet();
+      styleSheet.replaceSync(cssText);
+      cache[name] = styleSheet;
+    }
+    shadowRoot.adoptedStyleSheets = [styleSheet];
```

Comment:
Why overwrite rather than append?

Reply:
Good point. Switched to append. One caveat: it has to be a `concat` because the adopted stylesheets are refreshed only on setter. Because API something something.